ui/mirage/factories/service.js

```diff
@@ -19,16 +24,38 @@ export default Factory.extend({
       return null;
     }
   },
-  Connect: {
-    SidecarService: {
-      Proxy: {
-        Upstreams: [
-          {
-            DestinationName: dasherize(faker.hacker.noun()),
-            LocalBindPort: faker.random.number({ min: 5000, max: 60000 }),
-          },
-        ],
-      },
-    },
+
+  afterCreate(service, server) {
+    if (!service.namespace) {
+      const namespace = pickOne(server.db.jobs)?.namespace || 'default';
+      service.update({
+        namespace,
+      });
+    }
+
+    if (!service.node) {
+      const node = pickOne(server.db.nodes);
+      service.update({
+        nodeId: node.id,
+      });
+    }
+
+    if (server.db.jobs.findBy({ id: 'service-haver' })) {
+      if (!service.jobId) {
+        service.update({
+          jobId: 'service-haver',
+        });
+      }
```

Comment:
Should this logic be tied to the `namespace` assignment? Will this logic mix well with `allScenarios` in mirage?

Reply:
In testing re: allScenarios, so far so good.

I don't think it should be tied to the namespace assignment; it's in the same afterCreate block but otherwise separated.